src/main/java/org/apache/maven/buildcache/checksum/TreeWalkerPathMatcher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.buildcache.checksum;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+
+/**
+ * A path matcher with some extra info
+ */
+public class TreeWalkerPathMatcher implements PathMatcher {
+
+    /**
+     * True if the matching should stop exploring the directory tree further away
+     */
+    private final boolean matchingSkipSubtree;
+
+    /**
+     * Wrapped regular path matcher
+     */
+    private final PathMatcher pathMatcher;
```

Comment:
It should not, the underlying conversion from glob to regex pattern (sun.nio.fs.Globs#toRegexPattern) used by different FS pathmatcher implementations (ex : sun.nio.fs.WindowsFileSystem#getPathMatcher) seems to handle FS specific syntax conversion since java 8. (https://github.com/adoptium/jdk8u)

But PathMatching is not something I've dealt with a lot. Do you think we should take a belt and braces approach ? (french expression, but it looks like existing in English too ^^)

PS : On a side note, I'm always testing on Windows (but not all jvm ...) since I primarily work on Windows those days.

Reply:
There might be room for error here. For example, the same file with back/forward slashes is not matched on my workstation:
```java
        String linux = "dir/file.txt";
        String win = "dir\\file.txt";
        Path linuxPath = Paths.get(linux);
        Path winPath = Paths.get(win);
        System.out.println("linuxPath: " + linuxPath);
        System.out.println("winPath: " + winPath);
        PathMatcher linuxMatcher = FileSystems.getDefault().getPathMatcher("glob:" + linuxPath);
        System.out.println(linuxMatcher.matches(winPath));
        System.out.println(linuxMatcher.matches(linuxPath));
        ------
        false
        true
```
I have no objections against the matcher as such. But I wouldn't rely on it without a test demonstrating the same results on Windows and Mac in the presence of exclusions.